File: DSA/NiharikaChand_590017697/Day37Question1.java
import java.util.*;
class Day37Question1
{
    public static void sort(Stack<Integer> stack)
    {
        Stack<Integer> tempStk = new Stack<>();

        while (!stack.isEmpty())
        {
            int temp = stack.pop();
            while (!tempStk.isEmpty() && tempStk.peek() < temp)
            {
                stack.push(tempStk.pop());
            }
            tempStk.push(temp);
        }

        
        while (!tempStk.isEmpty())
        {
            stack.push(tempStk.pop());
        }
    }

    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        Stack<Integer> stack = new Stack<>();

        System.out.println("Enter number of elements:");
        int n = sc.nextInt();
       
        for (int i = 0; i < n; i++)
        {
             System.out.println("Enter an element:");
            stack.push(sc.nextInt());
        }

        sort(stack);
        System.out.println("Output" + stack);
    }
}

File: DSA/Nivedan_590014904/Day_37_Question_1.py
stack = []
n = int(input("ENter the number of elements to be put in the stack: "))
for i in range(n):
    val = int(input(f"Input the {i+1} element: "))
    stack.append(val)
temp = []
while stack:
    element = stack.pop()
    while temp and temp[-1] > element:
        stack.append(temp.pop())
    temp.append(element)
print(temp)
